docs/recipes/testing-with-selenium-webdriverjs.md

```diff
@@ -0,0 +1,131 @@
+
+```js
+test('Bing Search', async t => {
+	let keyword = "webdriver";
+	let driver = new Builder().forBrowser("chrome").build();
+	await driver.get("http://www.bing.com");
+	await driver.findElement(By.name("q")).sendKeys(keyword + Key.ENTER);
+	await driver.wait(until.titleIs(keyword + " - Bing"));
+	t.true((await driver.findElements(By.css("#b_content #b_results li"))).length > 0);
+	await driver.close();
+}); 
+```
+
+In the `googletest.js` file, instead of a single test, lets add two tests, one each for the terms 'webdriver' and 'avajs'.
+
+Since we would like to initialize the webdriver before each test, we use the [`beforeEach` and `afterEach`](https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#before--after-hooks) hooks to setup and teardown the driver respectively. Using these hooks, helps reduce the amount of code we would write in each test()
+
+```js
+test.beforeEach(async t => {
+	t.context.driver = new Builder().forBrowser("chrome").build();
+	await t.context.driver.get("http://www.google.com");
+});
+
+test.afterEach("cleanup", async t => await t.context.driver.close());
+```
+
+Now lets add the test code,
+```js
+test('Google Search for avajs', async t => {
+	let driver = t.context.driver;	
+	await searchGoogle(driver, "avajs");
+	t.true((await driver.findElement(By.id("resultStats")).getText()).includes("results"));
+});
+
+test('Google Search for webdriver', async t => {
+	let driver = t.context.driver;
+	await searchGoogle(driver, "webdriver");
+	t.true((await driver.findElement(By.id("resultStats")).getText()).includes("results"));
+});
+
+async function searchGoogle(driver, keyword) {
+	await driver.findElement(By.name("q")).sendKeys(keyword + Key.ENTER);
+	await driver.wait(until.titleIs(keyword + " - Google Search"));
+}
+```
+
+## Running the tests
+
+We can run all the tests, by the following command
+
+```console
+npm test
+```
+
+Since I ran the above command on a laptop with 4 CPU cores, AVA ran both `bingtest.js` and `googletest.js` in parallel. See below output.
+
+```console
+DevTools listening on ws://127.0.0.1:49852/devtools/browser/adfcad21-9612-46ff-adc3-09adc0737f4a
+
+DevTools listening on ws://127.0.0.1:49853/devtools/browser/304aab40-c81e-4f26-b19c-5616472d568a
+
+DevTools listening on ws://127.0.0.1:49855/devtools/browser/8f6b7206-ea2b-4d41-b6aa-10a42a562387
+
+  3 tests passed
+```
+
+Dependending upon the number of CPU cores, those many test files will run at the same time. This count can be changed either via terminal (--concurrency or -c) or via the `"ava"` section in the package.json. For e.g.
+
+```js
+"ava":{    
+    "concurrency": 1,
+    "verbose": true
+  }
+```
+
+The `concurrency: 1` value will only allow AVA to run one file at a time. It however cannot control how many tests in that file can run at the same time.
+
+The `verbose: true` value will enable verbose output which helps readability IMO.
```

Comment:
Sure, but let's keep opinion out of this :smile:

Reply:
Understood. Removed opinion.